From b735ac35cadb820f44fc34f3ee162996c4597a5d Mon Sep 17 00:00:00 2001
From: Jean Trivelly <jean.trivelly@intel.com>
Date: Thu, 13 Dec 2012 11:57:04 +0100
Subject: [PORT FROM MAIN] wifi: Unable to switch WiFi OFF with BCM.

BZ: 74777

After having enabled P2P for Broadcom solution,
It is no more possible to switch WiFi OFF.
Just a mutex unlock was missing.

Feature   : Wlan
Category  : Improvement
Purpose   : AOSP+
Origin    : Upstream
Upstream-candidate: yes

Change-Id: I708739acc23c9593b62ddb494a2b1d52a053366a
Orig-Change-Id: I6c200527b7c9374d635de47805f42fdcb01ef87e
Signed-off-by: Jean Trivelly <jean.trivelly@intel.com>
---
 wifi/wifi.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/wifi/wifi.c b/wifi/wifi.c
index 9e687b8..c076b89 100644
--- a/wifi/wifi.c
+++ b/wifi/wifi.c
@@ -906,6 +906,7 @@ void wifi_close_supplicant_connection(const char *ifname)
         /* p2p sockets are closed after the monitor thread
          * receives the terminate on the exit socket
          */
+        pthread_mutex_unlock(&suppl_mutex);
         return;
     }
 
-- 
1.7.4.1

